R/search_tweets.R

```diff
@@ -187,247 +187,68 @@ search_tweets <- function(q, n = 100,
-        error = function(e) return(NULL))
-      ## break if error
-      if (is.null(rt[[i]])) break
-      ## break if final i
-      if (i == ntimes) break
-      if (parse) {
-        ## get next maxid
-        maxid.new <- rt[[i]][["status_id"]][[NROW(rt[[i]])]]
-      } else {
-        maxid.new <- return_last(unlist(go_get_var(rt[[1]], "statuses", "id")))
-      }
-      ## break if new maxid is null, empty, or unchanged
-      if (any(is.null(maxid.new),
-        identical(length(maxid.new), 0L),
-        identical(maxid, maxid.new))) break
-      ## update maxid value
-      maxid <- maxid.new
-      ## refresh rate limit data
-      rtlimit <- rate_limit(token, "search/tweets")
-      remaining <- rtlimit[["remaining"]] * 100
-      reset <- rtlimit[["reset"]]
-      reset <- as.numeric(reset, "secs")
-    }
-    ## get users data if applicable
-    users <- do.call("rbind", lapply(rt, users_data))
-    rt <- do.call("rbind", rt)
-    attr(rt, "users") <- users
+  if (parse) {
+    result <- tweets_with_users(result)
   }
-  rt
+  result
 }
 
-
-.search_tweets <- function(q,
-                           n = 100,
-                           check = FALSE,
-                           geocode = NULL,
-                           type = "recent",
-                           max_id = NULL,
-                           include_rts = TRUE,
-                           parse = TRUE,
-                           token = NULL,
-                           verbose = TRUE,
-                           ...) {
-  ## gotta have ut8-encoding for the comma separated IDs
-  op <- getOption("encoding")
-  on.exit(options(encoding = op), add = TRUE)
-  options(encoding = "UTF-8")
-
-  ## path name
-  query <- "search/tweets"
-  safedir <- NULL
-  if ("premium" %in% names(list(...)) &&
-      all(c("env_name", "path") %in% names(list(...)$premium))) {
-    premium <- list(...)$premium
-    premium$path <- sub("tweets/search/?|search/tweets/?", "", premium$path)
-    query <- gsub("/+", "/",
-      paste0("tweets/search/", premium$path, "/", premium$env_name))
-    cat(query, "***")
-
-    if ("safedir" %in% names(list(...))) {
-      safedir <- list(...)$safedir
-    }
+search_params <- function(q, n, 
+                          type = c("mixed", "recent", "popular"),
+                          include_rts = TRUE,
+                          geocode = NULL,
+                          max_id = NULL,
+                          ...) {
+  if (missing(q) && !is.null(geocode)) {
+    q <- ""
   }
-  ## validate
   stopifnot(is_n(n), is.atomic(q), length(q) == 1L, is.atomic(max_id))
-  ## number of loops
-  n.times <- ceiling(n / 100)
-  if (n < 100) {
-    count <- n
-  } else {
-    count <- 100
-  }
+  type <- match.arg(type)
+  
   ## validate query length–char count might not always be same here as with 
   ## Twitter, so set this to 600 and let Twitter reject others
   if (nchar(q) > 600) {
     stop("q cannot exceed 500 characters.", call. = FALSE)
```

Comment:
There's a mismatch between the error message and the nchar used to trigger it.

Reply:
This is unrelated to the thrust of this PR, so it's better to resolve elsewhere.